src/instruction/mod.rs

```diff
@@ -132,3 +132,20 @@ impl Clone for Box<dyn Instruction> {
         self.box_clone()
     }
 }
+
+use inkwell::OptimizationLevel;
+use inkwell::builder::Builder;
+use inkwell::context::Context;
+use inkwell::execution_engine::{ExecutionEngine, JitFunction};
+use inkwell::module::Module;
+use inkwell::targets::{InitializationConfig, Target};
+
+pub struct CodeGen<'ctx> {
+    context: &'ctx Context,
+    module: Module<'ctx>,
+    builder: Builder<'ctx>,
+}
+
+pub trait Compilable {
+    fn ir(&self, ctx: &mut CodeGen, interpreter: &Interpreter) -> Result<String, JkError>;
+}
```

Comment:
I think this is a bit outside the scope of this PR but that's okay. However, careful of unused code.

The issue is that I believe this code should reside in its own file, but I'm not sure of what architecture to use. I think using another module would be best

Reply:
oops this shouldn't have been pushed